game_patch/multi/network.cpp

```diff
@@ -728,33 +759,134 @@ CallHook<int(const rf::NetAddr*, std::byte*, size_t)> send_join_accept_packet_ho
+    AlpineFactionRemoteInfo remote_info{};
+    remote_info.version_major = ext.version_major;
+    remote_info.version_minor = ext.version_minor;
+    using Flags = AlpineFactionJoinAcceptPacketExt::Flags;
+    remote_info.saving_enabled = !!(ext.flags & Flags::saving_enabled);
+    remote_info.allow_fb_mesh = !!(ext.flags & Flags::allow_fb_mesh);
+    remote_info.allow_lmap = !!(ext.flags & Flags::allow_lmap);
+    remote_info.allow_no_ss = !!(ext.flags & Flags::allow_no_ss);
+    remote_info.no_player_collide = !!(ext.flags & Flags::no_player_collide);
+    remote_info.allow_no_mf = !!(ext.flags & Flags::allow_no_mf);
+    remote_info.click_limit = !!(ext.flags & Flags::click_limit);
+    remote_info.unlimited_fps = !!(ext.flags & Flags::unlimited_fps);
+    remote_info.gaussian_spread = !!(ext.flags & Flags::gaussian_spread);
+    remote_info.location_pinging = !!(ext.flags & Flags::location_pinging);
+    constexpr const float default_fov = 90.f;
+    if (!!(ext.flags & Flags::max_fov) && ext.max_fov >= default_fov) {
+        remote_info.max_fov.emplace(ext.max_fov);
+    }
+    if (remote_info.click_limit) {
+        remote_info.semi_auto_cooldown.emplace(ext.semi_auto_cooldown);
+    }
+    g_af_remote_info.emplace(remote_info);
+    return true;
+}
+
 CodeInjection process_join_accept_injection{
     0x0047A979,
-    [](auto& regs) {
-        std::byte* packet = regs.ebp;
-        auto ext_offset = regs.esi + 5;
-        DashFactionJoinAcceptPacketExt ext_data;
-        std::copy(packet + ext_offset, packet + ext_offset + sizeof(DashFactionJoinAcceptPacketExt),
-            reinterpret_cast<std::byte*>(&ext_data));
-        xlog::debug("Checking for join_accept DF extension: {:08X}", ext_data.df_signature);
-        if (ext_data.df_signature == DASH_FACTION_SIGNATURE) {
-            DashFactionServerInfo server_info;
-            server_info.version_major = ext_data.version_major;
-            server_info.version_minor = ext_data.version_minor;
-            xlog::debug("Got DF server info: {} {} {}", ext_data.version_major, ext_data.version_minor,
-                static_cast<int>(ext_data.flags));
-            server_info.saving_enabled = !!(ext_data.flags & DashFactionJoinAcceptPacketExt::Flags::saving_enabled);
-
-            constexpr float default_fov = 90.0f;
-            if (!!(ext_data.flags & DashFactionJoinAcceptPacketExt::Flags::max_fov) && ext_data.max_fov >= default_fov) {
-                server_info.max_fov = ext_data.max_fov;
-            }
-            g_df_server_info = std::optional{server_info};
+    [] (const auto& regs) {
+        const std::byte* const packet = regs.ebp;
+        const int ext_offset = regs.esi + 5;
+        const std::byte* const tail = packet + ext_offset;
+        if (!parse_df_join_accept_tail(tail)) {
+            parse_af_join_accept_tail(tail);
         }
-        else {
-            g_df_server_info.reset();
+    }
+};
+
+ConsoleCommand2 dbg_remote_flags_cmd{
+    "dbg_remote_flags",
```

Comment:
To be honest this could be usable to normal users to check server settings, not just for debugging. 
Maybe it would be better to name it `server_flags` or something like that and make labels more user-friendly, e.g. replace `max_fps` with `Max FPS`, etc.
What do you think?